portfolio/src/main/webapp/script.js

```diff
@@ -52,3 +52,64 @@ function prev(){
 
 var myVar = setInterval(next, 4000);
 
+
+
+/**
+ * Fetches a message from the server and adds it to the DOM.
+ */
+function getMessage() {
+  console.log('Fetching a message.');
+
+  // The fetch() function returns a Promise because the request is asynchronous.
+  const responsePromise = fetch('/data');
+
+  // When the request is complete, pass the response into handleResponse().
+  responsePromise.then(handleResponse);
+}
+
+/**
+ * Handles response by converting it to text and passing the result to
+ * addQuoteToDom().
+ */
+function handleResponse(response) {
+  console.log('Handling the response.');
+
+  // response.text() returns a Promise, because the response is a stream of
+  // content and not a simple variable.
+  const textPromise = response.text();
+
+  // When the response is converted to text, pass the result into the
+  // addQuoteToDom() function.
+  textPromise.then(addMessageToDom);
+}
+
+/** Adds a message to the DOM. */
+function addMessageToDom(message) {
+  console.log('Adding message to dom: ' + message);
+
+  const messageContainer = document.getElementById('the-message');
+  messageContainer.innerText = quote;
```

Comment:
I don't think `quote` has meaning here. Maybe I'm missing something but it doesn't seem to be a parameter to the function nor a local variable. Am I confused?